src/integer/mat_z/inverse.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright © 2023 Marcel Luca Schmidt
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains the implementation of the `inverse` function.
+
+use super::MatZ;
+use crate::{
+    error::MathError,
+    integer::Z,
+    rational::MatQ,
+    traits::{GetNumColumns, GetNumRows},
+};
+use flint_sys::fmpq_mat::fmpq_mat_inv;
+
+impl MatZ {
+    /// Returns the inverse of the matrix if it exists (is square and
+    /// has a determinant unequal to zero).
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::integer::MatZ;
+    /// use qfall_math::traits::*;
+    /// use std::str::FromStr;
+    ///
+    /// let mut matrix = MatZ::from_str("[[1,2],[3,4]]").unwrap();
+    /// let matrix_invert = matrix.invert().unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`MismatchingMatrixDimension`](MathError::MismatchingMatrixDimension)
+    /// if the number of rows and columns is not equal.
+    /// - Returns a [`MathError`] of type [`NotInvertible`](MathError::NotInvertible)
+    /// if the determinant of the matrix is `0`.
+    pub fn invert(&self) -> Result<MatQ, MathError> {
```

Comment:
For `Q` the invert function is called `inv`. Please apply the same naming scheme here.